src/components/CheckEditor/FormComponents/HttpCheckAuthentication.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { GrafanaTheme2 } from '@grafana/data';
+import { RadioButtonGroup, useStyles2 } from '@grafana/ui';
+import { css } from '@emotion/css';
+
+import { CheckFormValuesHttp, HttpAuthType } from 'types';
+
+import { HttpCheckBasicAuthorization } from './HttpCheckBasicAuthorization';
+import { HttpCheckBearerToken } from './HttpCheckBearerToken';
+
+export const HttpCheckAuthentication = () => {
+  const styles = useStyles2(getStyles);
+
+  const { getValues, setValue } = useFormContext<CheckFormValuesHttp>();
+
+  const isBasicAuth = Boolean(
+    getValues(`settings.http.basicAuth.username`) || Boolean(getValues(`settings.http.basicAuth.password`))
+  );
+
+  const isBearerAuth = Boolean(getValues(`settings.http.bearerToken`));
+
+  const authType = getValues('settings.http.authType');
+
+  useEffect(() => {
+    if (isBasicAuth) {
+      setValue('settings.http.authType', 'basic');
+      return;
+    }
+    if (isBearerAuth) {
+      setValue('settings.http.authType', 'bearer');
+      return;
+    }
+    setValue('settings.http.authType', 'none');
+  }, [isBasicAuth, isBearerAuth, setValue]);
```

Comment:
I don't think this needs a `useEffect`. It's effectively only used on initial load as the state it is handling is taken over by `handleChangeOption` so could be derived from the existing state instead.